txpool/account.go

```diff
@@ -57,7 +58,7 @@ func (m *accountsMap) getPrimaries() (primaries []*types.Transaction) {
 		defer account.promoted.unlock()
 
 		// add head of the queue
-		if tx := account.promoted.peek(); tx != nil {
+		if tx := account.promoted.peek(); tx != nil && tx.GetGasFeeCap().Cmp(new(big.Int).SetUint64(baseFee)) >= 0 {
```

Comment:
what do you think to move this check `tx.GetGasFeeCap().Cmp(new(big.Int).SetUint64(baseFee))` to some function inside `transaction.go` - something like `CheckGasFeeCap(baseFee) bool`? then you can replace calculations on bunch of places. and maybe have `*big.Int` as parameter instead of `uint64` (later on we can change `baseFee` to `*big.Int `on other places)

Reply:
This condition just used twice and it will be even simpler after EIP-2718. I think we should not extract this to a function.